PowerEditor/src/WinControls/OpenSaveFileDialog/CustomFileDialog.cpp

```diff
@@ -311,20 +315,19 @@ class FileDialogEventHandler : public IFileDialogEvents, public IFileDialogContr
 	{
 		if (id == IDC_FILE_TYPE_CHECKBOX)
 		{
-			bool ok = false;
+			UINT newFileType = 0;
 			if (bChecked)
 			{
-				ok = SUCCEEDED(_dialog->SetFileTypeIndex(_lastSelectedType));
-				ok &= OnTypeChange(_lastSelectedType);
+				newFileType = _lastSelectedType;
 			}
 			else
 			{
-				UINT currentIndex = 0;
-				ok = SUCCEEDED(_dialog->GetFileTypeIndex(&currentIndex));
-				if (ok && currentIndex != 0 && currentIndex != _wildcardType)
-					_lastSelectedType = currentIndex;
-				ok &= SUCCEEDED(_dialog->SetFileTypeIndex(_wildcardType));
+				if (_currentType != 0 && _currentType != _wildcardType)
+					_lastSelectedType = _currentType;
+				newFileType = _wildcardType;
 			}
+			_dialog->SetFileTypeIndex(newFileType);
```

Comment:
Should there be a sanity check just to make sure that `newFileType` is not the same as `_currentType`? That is, that it actually changed?

Reply:
Good point.
But I believe, it could never happen.

This function is called each time the checkbox is clicked.
`_lastSelectedType` could be anything but `_wildcardType` (condition at the line 325).
And when the new file type is selected (user clicks an item in the dropdown list) the checkbox is checked if it was unchecked.